### DISCUSSION TITLE
数式のプレビューがおかしい

### DISCUSSION THREAD

**Discussion by tsubolabo** [increments/qiita-discussions#298] [要望・アイデア・不具合]
いつもお世話になっております。

投稿エディタのプレビュー画面で、数式 $\mathbf{e}\_x$ などの表示が不安定です。画像を添付します。プレビューで確認しているだけなので、正式投稿画面でも起こる現象なのかは分かりません。

<img width="959" alt="Qiita投稿画面" src="https://user-images.githubusercontent.com/42261605/190095560-fa1e864d-31db-4a0b-a0aa-1b7c29eb0441.png">

４つの例を示しています。本来は、１つ目の記述で３つ目の出力が表示されるはずなのですが、うまく行きません。

下付き文字の前の "\_" は、そのままでも問題ない部分もあれば、 "\_" の代わりに "\\\_" としないと正常に表示されない部分もあります。しかし、 "\\\_" でも受け付けないこともあります。

従来でも、"\left(" や "\left." などは、 "\left\\(" や "\left\\." などとエスケープが必要なことは認識していましたが、下付き文字用のアンダーバーはエスケープせずに正常に動作していました。今回のような問題は初体験です。

使用環境： windows10,  microsoft Edge（chromium）

**Comment by kskwtnk** [comment]
ご意見ありがとうございます！

数式関連のご意見は他でもいただいており、対応できておらずすみません 🙇 
調査や解決がなかなか思うようにいかずお時間をいただいています。

こちらも現在の調査中のリストに入れて作業を進めますが、何卒ご了承ください 🙇

**Reply by tsubolabo** [reply]
宜しくお願い致します。
また、こちらも一緒に御検討いただければ幸いです。

１．不必要なスクロールバー（私としては、これが最優先課題）
２．ベースラインの不揃い
３．文字の上の装飾記号の隠れ

なお、ブラウザの拡張機能の Markdown Viewer では、\smash[t] のお世話にならなくても、ベースラインは綺麗に揃います。また、装飾記号も、行間を保ったまま、隠れずに表示されます。できれば、これに合わせた特性を期待します。

<img width="960" alt="Qiita投稿画面02" src="https://user-images.githubusercontent.com/42261605/190311583-86bffc48-06d6-4313-8676-1a0c82c6335c.png">

**Comment by kyntk** [comment]
@tsubolabo
https://github.com/increments/qiita-discussions/discussions/498 でもお伝えしましたが、MathJaxのバージョンアップを行いました。

上記ご報告いただいた内容については、このバージョンアップで改善される見込みなのですが、何か変化はあるでしょうか？
お手すきの際にご確認いただけると幸いです🙏

**Reply by tsubolabo** [reply]
@kyntk 

このディスカッションの中では、下記の４件の不具合を挙げていました。

①　数式の下付き添え字用の"_"の不規則な動作
②　数式の下に邪魔なスクロールバーが現れる
③　インライン数式と本文のベースラインの落差
④　数式文字の上部の装飾記号が隠れて見えない

メインテーマだった①は全く改善されていません。むしろ、今回のV3化に伴うベースラインのズレまで加わり、さらに悪くなっています。

②～④は、既に以前（2022-11-15～）に解決された問題だったのに、今回のV3化により、一部は悪化して逆戻りです。
②については、今回の変更で逆戻りしたという様子は見られません。
しかし、③については、#498で指摘されているように、逆戻りというよりも以前よりも酷くなっています。以前に存在したベースラインの問題は、特殊な装飾を施した数式だけに起こるものでしたが、今回の不具合は条件を選ばずどんな単純な数式にも発生します。その影響は甚大です。
今の状態では、④まで評価できる段階ではありません。

現状での私の希望は#425のコメントに書いています。

**Reply by kyntk** [reply]
@tsubolabo ご確認いただきありがとうございます :pray:

①について
こちらでも解消されていないことが確認されました。
原因はMarkdownの記法と衝突しているのが原因のようなので、 `_` の前後にスペースを入れていただくと解消しました :bow: 
https://kmyk.github.io/blog/blog/2020/02/19/portable-mathjax-markdown/

②、③についてはこちらでも引き続き確認してみます。
④については、もし再現した場合、その数式とスクリーンショットをお送りいただけますと幸いです :pray:

**Reply by tsubolabo** [reply]
@kyntk 

>今の状態では、④まで評価できる段階ではありません。
>> ④については、もし再現した場合、その数式とスクリーンショットをお送りいただけますと幸いです 🙏

「今の状態」とは、#498の問題が解決されていない状態のことです。
まずは、#498の問題の早急なる改善をお願いします。

**Reply by tsubolabo** [reply]
@kyntk 

訂正です。前回の報告は、投稿エディタの中だけで確認した結果によるコメントでした。しかし、自身が過去に投稿した実際の記事上で確認したところ、②についての下記のコメントは誤まりでした。スクリーンショットに示すように、とんでもないほど悪化しています。インライン数式だけでなく、別行建て数式にさえ余計なスクロールバーが出現しています。
>②については、今回の変更で逆戻りしたという様子は見られません。


<img width="562" alt="数式01" src="https://github.com/increments/qiita-discussions/assets/42261605/525179cc-2856-42bc-8215-4c25841b50de">

<img width="556" alt="数式02" src="https://github.com/increments/qiita-discussions/assets/42261605/23e52ff5-0fc0-412d-8e31-c84461cf909f">
　　　
　　　
　また、下記は#498の問題に関連したものですが、F=F+F なんとかかんとか　のインラインの式が右肩上がりのマヨビーム風になっています。問題解決のヒントになりそうなので追加しておきます。

<img width="556" alt="数式03" src="https://github.com/increments/qiita-discussions/assets/42261605/fe1a9036-784d-4e43-aed3-5fb3fb03a0d1">

**Reply by kyntk** [reply]
@tsubolabo 度々詳しく調査いただきましてありがとうございます。:pray: 
完全に解決するのは難しそうだったのですが、②、③に関して少し改善を加えてみました。

お使いのブラウザによっては改善が見られない可能性もありますが、いかがでしょうか?
ご確認いただけますと幸いです。:bow:

**Reply by tsubolabo** [reply]
@kyntk 

>完全に解決するのは難しそうだったのですが、②、③に関して少し改善を加えてみました。
>ご確認いただけますと幸いです。

その前に、新たに見つかった問題点の報告です。2023-07-14昼現在の現象なので、「改善」を加えたあとですが、07-06の変更により既に生じていたものなのか？　07-13の改善によるものなのか？　は判別できません。**テーブルの右横に無駄な縦スクロールバーが表示され、スクロールしないと全体が見られないようになりました。**

<img width="555" alt="数式04" src="https://github.com/increments/qiita-discussions/assets/42261605/aae9ad35-d790-483a-85c6-ce867e045f90">

<img width="616" alt="数式05" src="https://github.com/increments/qiita-discussions/assets/42261605/8386d6fc-e778-4d75-95b3-2ca8cb7a5045">

さて、本題に戻ります。

前回のコメントに添付したスクリーンショットと同一の箇所での「改善」前後の比較です。最初が「前」、次が「後」です。「前」に比べると、「なかなかうまいこと丸め込んだな」という印象で赤枠をオレンジ枠に変えていますが、まだまだです。

<img width="562" alt="数式01" src="https://github.com/increments/qiita-discussions/assets/42261605/0a23591f-8a60-4508-99a0-49ddecdb85f9">

<img width="554" alt="数式07" src="https://github.com/increments/qiita-discussions/assets/42261605/ee9e92ea-c079-4bc4-bb4d-1f05646cef47">

この画面では数式下のスクロールバーは消えましたが、一部では未だ次のように残っています。

<img width="554" alt="数式06" src="https://github.com/increments/qiita-discussions/assets/42261605/db21293d-bf91-49cf-ab81-fa0fe7888710">

また、θ などの上のドットの位置が上がり過ぎて、醜くなっています。これはインライン数式だけでなく、別行立て数式にまで発生するようになりました。次の赤枠のスクリーンショットは Qiita のもの、その次の緑枠は Chrome の拡張機能（Markdown Viewer、About MathJax で見ると MathJax v3.2.2 が使われているようです）のものです。ver2 当時の Qiita の数式はこれと同じようにドットの位置は適正でした。

<img width="557" alt="数式08" src="https://github.com/increments/qiita-discussions/assets/42261605/1e50529a-6c48-4f21-8671-b385c714f6aa">

<img width="535" alt="数式09" src="https://github.com/increments/qiita-discussions/assets/42261605/ff47f6c0-2500-4606-ae92-e1e77c2405e2">

良い方向に行っているとは思いますが、適当な所まで持って行って「これは仕様です」と突き放されそうで油断なりません。Qiita の中に、「こんな数式では会社の品位や評価が落ちる」と思ってくださる方が居られることを祈ります。

>完全に解決するのは難しそう（再掲）

MathJax v3 で Chrome の Markdown Viewer ができていることを、Qiita ができないとは、諦めが早すぎるのではありませんか？

**Reply by kyntk** [reply]
@tsubolabo
追加で詳しい情報をお送りいただきありがとうございます。
調査したのですが、修正にはまだ時間がかかりそうです。
直近の開発優先度などから考えると対応は先になってしまうと思いますが、修正していきたいと思います :pray:

**Reply by tsubolabo** [reply]
最後に頂いた回答からそろそろ１年になります。数式表示の問題については鋭意改善中のことと思われます。大変お世話をかけております。

　そのような中で問題点ばかり増やして申し訳ありませんが、新たに見つかった問題を下記に画像として示しています。数式の中身は同じですが囲み記号が ` ```math` か `$$` かで異なるだけです。 ` ```math` 方式の方の表示は完全にアウトです。

<img width="532" alt="数式10" src="https://github.com/increments/qiita-discussions/assets/42261605/33e9e584-e124-46d8-aee3-eaa5148f4024">

　
　ところで、Qiita の「Markdown 記法チートシート」によれば、数式の正式な記法は上図の上の方法です。しかし、一般には下の方法が主流です。私もすべての投稿で下の方法を使用しているのですが、これによる問題は感じたことはありません。この際、下の方法を正式な記法として採用しては如何でしょうか？　現在の方法を無効にして欲しい訳ではありませんが、数ある数式の問題を一つの記法に絞って片づけていく方が効率が良さそうですので・・・。

**Reply by tanaka-asami415** [reply]
@tsubolabo 
ご意見ありがとうございます！未だ解決には至っておらず申し訳ありません。

> 一般には下の方法が主流です。私もすべての投稿で下の方法を使用しているのですが、これによる問題は感じたことはありません。

現状では、` ```math`  であれば MathJax の記法で書いていただければ基本問題ないのですが、
`$$` の場合は Markdown のエスケープを意識して書いていただく必要があります。

そのため、現状は正式な記法として採用することは難しいですが、`$$` の記法も可能であることを記載するなどは、検討したいと思います！

以下の例だと、全く同じ入力でも、下はアンダースコアのエスケープをしていないので崩れます。

````
```math
\mathbf{e}_r \mathbf{e}_\theta
```
````

```
$$
\mathbf{e}_r \mathbf{e}_\theta
$$
```

tsubolaboさんの式の場合、`1&(i=j)\\\\` を`1&(i=j)\\` とご記載いただければ、` ```math` でも綺麗に表示されるのではないかと思います。

````
```math
\begin{align}
  \delta_{ij}=
  \begin{cases}
    1&(i=j)\\
    0&(i\neq j)
  \end{cases}
\end{align}
```
````

**Reply by tsubolabo** [reply]
@tanaka-asami415 

> 現状では、 ```math であれば MathJax の記法で書いていただければ基本問題ないのですが、$$ の場合は Markdown のエスケープを意識して書いていただく必要があります。

　それは全く知りませんでした🙇。お詫びのしるしに訂正した画像を貼っておきます。なるほど、どちらも同じ数式になりました。別行立て数式の場合には math 方式は助かりますね（ただし、\dot{} の位置が高すぎるなどの数式の諸問題が解決するわけではないようです）。

　しかし、インライン数式では$方式しか使えず、それにはエスケープが必要になるというのでは、取り扱いが複雑化しているだけのように感じてしまいます。

<img width="500" alt="数式11" src="https://github.com/increments/qiita-discussions/assets/42261605/9f3d3cc4-480a-4bbc-8143-8d405c649a64">

　
> 未だ解決には至っておらず申し訳ありません。

　まだほとんど手付かずということでしょうか？　であれば、再度、数式表示の重要性を訴えさせていただきます。開発優先度の繰り上げに繋がれば良いのですが・・・。

　Qiitaは、「コミュニティガイドライン」の中で、（プログラマーだけではなく）すべてのエンジニアがお互いの記事を通じて成長する場を作ることを宣言されています。エンジニアの思考の拠り所は工学系の理論であり、その理論を曖昧性なく表現できるのは数式です。数式の表示に多くの問題を抱えながらエンジニア用のコミュニティを名乗るのには違和感があります。ブーイングの嵐を覚悟のうえで言わせていただけば、私にとって、数式表示問題の解決はダークモード化よりも遥かに重要です。

**Reply by tanaka-asami415** [reply]
@tsubolabo 
> 別行立て数式の場合には math 方式は助かりますね（ただし、\dot{} の位置が高すぎるなどの数式の諸問題が解決するわけではないようです）。

> しかし、インライン数式では$方式しか使えず、それにはエスケープが必要になるというのでは、取り扱いが複雑化しているだけのように感じてしまいます。

ご意見ありがとうございます。参考にさせていただきます🙇

特別に数式関連の優先度が低いと考えている訳ではないですが、
現在は、安定的な収益を作り出せる開発を優先している状況にあり、中々対応が追い付いておらず申し訳ありません。

持続的に既存機能の改善や新機能の開発を行い、Qiitaというサービスやコミュニティを維持するためには、そのためのコストが必要であり継続的な収益が必要不可欠となります。
そのため直近では、Qiita Engineer Festaや企業とのコラボレーションページなどの開発を優先しております。

もちろん、Discussions 等でいただいているご要望等も対応していきたいと考えていますが
開発の他に障害対応やお問合せ調査なども行っているため、どうしても時間がかかってしまいます。

サービスがより成長していければ、提供できるサービスの質を向上させることもできるようになるため、
Qiitaをより良いプラットフォームに進化させていくために、ご理解いただけると幸いです🙇

**Reply by tsubolabo** [reply]
@tanaka-asami415 

　なかなか大変ですね。Qiita の気迫を感じます。気長に待つことにします。

**Reply by IsseiNAKASONE** [reply]
同じ問題に直面しています。状況は把握しました。
（一つインライン中にアンダースコアが複数出るとコンパイルできないと理解しています。）
ご対応の方よろしくお願いしますm(_ _)m

**Reply by P-SiZK** [reply]
@tsubolabo @IsseiNAKASONE 
大変お待たせしました :bow:

Markdown のエスケープを意識せずにインライン数式が記述できるように、インライン数式に新しい記法 ``$`...`$`` を導入しました！
[リリースノートはこちら](https://qiita.com/release-notes/2025-09-30)

今後とも Qiita をよろしくお願いいたします :pray:

**Reply by tsubolabo** [reply]
改良をありがとうございます。しかし、私が予想していた対応方法とはやや異なるものだったため、コメントがしづらくお礼が遅れてしまいました。

改良への前向きな姿勢を評価しつつも、率直な感想を述べさせていただきます。

まずは、私の勝手な？期待：
　別行立て数式、インライン数式いずれの場合も、$$～$$ や \$ ～ \$ だけでエスケープなしの記述が可能になる。
（今、探しても見付けられないのですが、以前に他の方の質問に対する tanaka-asami415 さんのコメントに、「$$～$$ でも、\`\`\`math～\`\`\` なみに、エスケープなしでも対応できるようにしたい」というような内容があったと記憶しています。これに期待をかけていました。勘違いでしたら申し訳ありません。）

　今回の対応を拝見すると、別行立て数式は「\`\`\`math～\`\`\`」、インライン数式は「$\`～\`$」方式に限定することで問題は解決できたと考えておられるように見受けられます。そうであれば、チートシートに「$$～$$」の使用は推奨しない旨のコメントを追加されては如何でしょうか？　他のマークダウンでは「$$～$$」が使用できるため、ついつい惰性でこれを使ってしまいがちです。Qiita でも、多くのケースではそのまま通用するので、かえって、不具合に直面したときの不信感が大きくなってしまいます。

　以降は「\`\`\`math～\`\`\`」「$\`～\`$」の優先使用を心がけたいと思います。数式については、引き続き、次の問題点の改善にも取り組んで頂くようお願いします。（取り敢えず、今のところ頭にあるものだけを挙げてみました）

１．インライン数式のベースラインが不揃いになる。
２．文字の上の装飾記号の位置が高すぎる。
３．大型括弧（場合分けなど）が上下に分断する。
４．数式番号の上下方向の位置ずれ。

**Reply by TETSURO1999** [reply]
@tsubolabo さんへ

自分も以前から Qiita の数式表現 MathJax の動作には疑問を持っており、ご指摘の４点について自分でも再現を試みましたが如何でしょうか？

[Qiitaの数式表現MathJaxが何かおかしい  - Qiita](https://qiita.com/TETSURO1999/private/362da1edb54384fa98e4)

なお、OS および Web ブラウザなど環境も教えて頂けると幸いです。

**Reply by tsubolabo** [reply]
@TETSURO1999 さん

わざわざ分かりやすい実例をお示しいただきありがとうございます。まさしく、私が Qiita さんに改善をお願いしたい問題点そのものです。

なお、私の常用環境は、Windows11 ,Microsoft Edge です。
時々、Google Chrome や iPad, Androidスマホでも閲覧することもあります。

**Reply by tanaka-asami415** [reply]
@tsubolabo 
ご確認と丁寧なフィードバックをいただきありがとうございます🙏
こちらこそ、返信が遅くなってしまい申し訳ありません。

> １．インライン数式のベースラインが不揃いになる。
> ２．文字の上の装飾記号の位置が高すぎる。
> ３．大型括弧（場合分けなど）が上下に分断する。
> ４．数式番号の上下方向の位置ずれ。

以前からご要望をいただいている点について、引き続き対応時期をお約束できず申し訳ないですが、今後も改善に向けて検討を進めてまいります。

> （今、探しても見付けられないのですが、以前に他の方の質問に対する tanaka-asami415 さんのコメントに、「$$～$$ でも、\`\`\`math～\`\`\` なみに、エスケープなしでも対応できるようにしたい」というような内容があったと記憶しています。これに期待をかけていました。勘違いでしたら申し訳ありません。）

おそらく、こちらのコメントを指していただいているかと思います。
https://github.com/increments/qiita-discussions/discussions/955#discussioncomment-11795199

当初は、`$$〜$$` の改善も検討していましたが、Markdownパーサーが複雑化する影響も踏まえ、 ``$`...`$`` の形で提供させていただきました。
今回のアップデートにより、ブロック・インラインの両方でエスケープ不要の記法をご利用いただけるようになりましたので、今後は、ブロックには ` ```math〜``` `、インラインには ``$`...`$`` での記述を推奨させていただければと思います。
また、ご要望が多く寄せられた際には、改めて `$$～$$` の対応も検討させていただきます。

[Markdown記法チートシート](https://qiita.com/Qiita/items/c686397e4a0f4f11683d#%E6%95%B0%E5%BC%8F%E3%81%AE%E8%A8%98%E8%BF%B0%E6%96%B9%E6%B3%95)については、tsubolabo さんからのフィードバックをもとに内容を更新いたしました。
貴重なご意見をお寄せいただき、ありがとうございます🙇
引き続きよろしくお願いします。

**Reply by tsubolabo** [reply]
@tanaka-asami415 さん

> Markdown記法チートシートについては、tsubolabo さんからのフィードバックをもとに内容を更新いたしました。

せっかく更新いただいたのに申し訳ないのですが、ちょっと視点が違うような気がします。

私としては、JavaScript の中で MathJax を多用した経験もあり、シンプルな規則のエスケープには特に抵抗はありません。

Qiita で問題なのは、規則性が読み取れない複雑怪奇なエスケープが必要になる部分があることです。似たような記述の部分でも、エスケープなしで動作する場合もあれば、エスケープが必須のこともあります。現在、ユーザーとして問題を感じているのは、この複雑怪奇な部分です。

この複雑な部分には全く言及なしで、エスケープさえしておけば、$$～$$ や $ ～ $ でも問題なく使えると思わせるような説明を追加することは、かえって混乱を招きます。

ユーザーにとっては複雑と感じられる部分でも、プログラムの処理手順をご存じの Qiita さんであれば、それほど複雑な規則ではないと判断されているのかもしれません。もしそうであれば、ユーザーが泥沼に嵌らなくても済むように、もっと詳細なエスケープ規則を示していただければ、また話は変わってきます。

> 以前からご要望をいただいている点について、引き続き対応時期をお約束できず申し訳ないですが、今後も改善に向けて検討を進めてまいります。

あれからかなり待っている気がしますが、まだその程度の位置づけに留まっているのはチョット残念です。西九州新幹線のフル規格全線開通と同程度のお預け度を感じます。Qiita 様、佐賀県民様のお情けに縋るのみです。

**Reply by TETSURO1999** [reply]
@tsubolabo  さんに確認して頂きたいことがあります。

Qiita の MathJax は出力形式を選択できます。デフォルトは 軽量の CHTML 形式ですが，少々重いながらも正確な表示が可能とされる SVG 形式に切り替えることができます。具体的には，数式ブロックの上で右クリックするとメニューが出てきます。ここで Math Settings - Math Renderer で CHTML 形式と SVG 形式を選択できます。なお，この変更はページ全体に影響を及ぼすので，同一ページ内に複数の数式ブロックがある場合，どれか一つの設定を変更するだけで同じページ内にある他の数式ブロックにも適用されます。

これで不具合は改善されたように見えませんか？

なお，これはあくまで閲覧者のブラウザで表示モードを変更しただけに過ぎず，他の人の表示には影響を与えません。あくまで原因究明のための調査です。以下は，SVG 形式に変更すれば改善されたものとして話を進めます。

### 推定される不具合要因

指摘された不具合はすべて縦方向の表示ズレという点で共通しており，いずれも SVG 形式に切り替えると改善されます。Qiita は記事を見易くするため，比較的行間隔を広げた設定にしていますが，これが一部の MathJax の表示にも影響を及ぼしており，結果として数式全体の表示に不整合が生じているものと考えられます。SVG 形式に切り替えると数式の行間隔が縮むので，これが本来 MathJax で想定している行間隔ではないかと推察されます。

### Qiita 運営への要望事項

以下まとめます。

1. MathJax の CHTML 形式での表示が正常になるようにスタイルシート設定を見直す。
2. MathJax のデフォルト出力形式を CHTML から SVG に変更する。
3. 記事の作成者が MathJax のデフォルト出力形式を CHTML または SVG のいずれか選択できるようにする。

ちなみに 1 については自分も CSS ファイルを眺めてみましたが，どこをどう直して良いのかさっぱり分かりませんでした。おそらく技術的には 2 が一番容易なのではないかと推察します。もちろん副作用があるので決断が必要ですが 2 が可能であれば 3 は不要ですね。

### 一般利用者の対応策

PlantUML の Latex または ASCIIMath であれば表示は崩れません。ただし，改行位置が限られたり，数式番号が使えないなど MathJax と完全互換ではありません。

### 最後に

下記検証記事をまとめました。

https://qiita.com/TETSURO1999/items/30ff9035f8477cd2b4f9

**Reply by tsubolabo** [reply]
@TETSURO1999 さん

改善への熱意に感謝しております。

早速、自分の過去の記事で試してみました。ただし、エスケープ付きの \$\$ ～ \$\$、\$ ～ \$ 形式で書かれたままでのチェックです（残念ながら、まだ \`\`\`math ～ \`\`\` 形式では記事を書いたことがありません）。チェックした環境は Windows11 の Edge のみです。

### １．ベースラインの不揃い　および　２．装飾記号の位置が高い

２については、SVG 形式化により、文字と記号の相対位置関係は完璧に改善されています。しかし、同時にベースラインも変わってしまいます。それが良い方向に働く場合もあれば、逆効果のこともあります。

１については、SVG 形式化でベースラインは変化するものの、どちらが自然に見えるかはケースバイケースのような気がします。

### CHTML:
%=======================================
<img width="500" height="273" alt="chtml01" src="https://github.com/user-attachments/assets/42d0cdea-1a11-4cd0-bd01-c011dc513347" />
%=======================================

### SVG:
%=======================================
<img width="505" height="272" alt="svg01" src="https://github.com/user-attachments/assets/d6284bf0-cad0-4ed0-906d-99a0f494b4ef" />
%==========================================

### ３．大型括弧の上下分割

SVG 形式化により完璧に改善します。

### CHTML:
%=======================================
<img width="483" height="110" alt="chtml02" src="https://github.com/user-attachments/assets/4390b6e9-8cb0-483e-ba83-d83180782fb3" />
%=======================================

### SVG:
%=======================================
<img width="489" height="97" alt="svg02" src="https://github.com/user-attachments/assets/c661c819-a55e-4e25-aa41-02747d92eb6d" />
%=======================================

### ４．数式番号の上下位置のずれ

SVG 形式化により完璧に改善します。

### CHTML:
%=======================================
<img width="501" height="127" alt="chtml03" src="https://github.com/user-attachments/assets/590219fb-2ea9-4f5c-b1de-5eff261bd557" />
%=======================================

### SVG:
%=======================================
<img width="502" height="122" alt="svg03" src="https://github.com/user-attachments/assets/2f24cdfe-6d9b-4cbf-88be-9e89b3f33304" />
%=======================================

**Reply by TETSURO1999** [reply]
@tsubolabo さん

なるほど！ベースライン問題は SVG 化だけでは解決できないかもしれませんか・・・これは根が深そうですね。実は気づいた点がもう一つあります。たとえば以下の数式ブロックです。

```
$$ \large
\dot{e} = \dot{\mathbf{e}} = \dot{\mathbb{e}}
$$
```

ココの github だと黒板太字 $\large\mathbb{e}$ も含めてドット記号は<ruby>いい<rt>●●</rt></ruby>感じに配置されます。
※ブラウザ依存性があり，Edge だと太字と黒板太字そのものが表示されません。Firefox だと正常に表示されます。

$$ \large
\dot{e} = \dot{\mathbf{e}} = \dot{\mathbb{e}}
$$

しかし，Qiita だと結構離れてしまうんですよね。以下は Qiita の画面（Windows11, Microsoft Edge）のスクリーンショット画像ですが，下記は出力形式が CHTML の場合です。

<img src="https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/3490655/4c7f8b38-c854-4b4e-ba46-78e958aa8e52.png">

下記は出力形式を SVG にしたもの（の同じくスクリーンショット画像）ですが，黒板太字だけは少し離れています。

<img src="https://qiita-image-store.s3.ap-northeast-1.amazonaws.com/0/3490655/414ba015-c25f-4913-81da-6daf9854cb1d.png">

zenn も note も KaTeX を採用しているようなので，MathJax を使用していて直接比較できそうなのは github くらい？
ということで Qiita と github の css 設定の何が違うのか見比べているところです。github だとカッコは綺麗に繋がって見えますが，数式番号がNGですね。
※ブラウザ依存性があり，Firefox だと数式番号自体表示されません。Edge だと数式番号がカッコの左側に表示されます。

$$
\left \\{ \begin{eqnarray}
   a! \equiv 0    \pmod 3 \\
-4m^2 \equiv -m^2 \pmod 3 \\
  4^c \equiv 1    \pmod 3 \\
 2024 \equiv 2    \pmod 3
\end{eqnarray} \right. \tag{6}
$$

```
$$
\left \\{ \begin{eqnarray}
   a! \equiv 0    \pmod 3 \\
-4m^2 \equiv -m^2 \pmod 3 \\
  4^c \equiv 1    \pmod 3 \\
 2024 \equiv 2    \pmod 3
\end{eqnarray} \right. \tag{6}
$$
```

**Reply by ohakutsu** [reply]
@tsubolabo 
ご意見ありがとうございます。

いただいたフィードバックをもとに、Markdown記法チートシートを再度更新させていただきました。
https://qiita.com/Qiita/items/c686397e4a0f4f11683d#%E6%95%B0%E5%BC%8F%E3%81%AE%E8%A8%98%E8%BF%B0%E6%96%B9%E6%B3%95

かなりおまたせしており大変申し訳ないのですが、今後も改善に向け検討させていただきます。
